crates/holochain/tests/tests/publish/mod.rs

```diff
@@ -29,56 +31,85 @@ async fn publish_terminates_after_receiving_required_validation_receipts() {
 
     let ((alice,), (bobbo,), (carol,), (danny,), (emma,), (fred,)) = apps.into_tuples();
 
-    let action_hash: ActionHash = conductors[0]
-        .call(&alice.zome(TestWasm::Create), "create_entry", ())
-        .await;
+    let apps = [alice, bobbo, carol, danny, emma, fred];
 
-    // Wait until they all see the created entry, at that point validation receipts should be getting sent soon
-    await_consistency(60, [&alice, &bobbo, &carol, &danny, &emma, &fred])
-        .await
-        .unwrap();
-
-    let ops_to_publish = tokio::time::timeout(std::time::Duration::from_secs(60), async {
-        let mut ops_to_publish = 1;
-        while ops_to_publish > 0 {
-            ops_to_publish = alice
-                .authored_db()
-                .read_async({
-                    let alice_pub_key = alice.agent_pubkey().clone();
-                    // Note that this test is relying on this being the same check that the publish workflow uses.
-                    // If this returns 0 then the publish workflow is expected to suspend. So the test isn't directly
-                    // observing that behaviour but it's close enough given that there are unit tests for the actual
-                    // behavior.
-                    move |txn| num_still_needing_publish(txn, alice_pub_key)
-                })
-                .await
-                .unwrap();
-        }
+    for c in conductors.iter() {
+        c.declare_full_storage_arcs(apps[0].dna_hash()).await;
+    }
+
+    // wait for all our conductors to see each other
+    tokio::time::timeout(std::time::Duration::from_secs(60), async {
```

Comment:
Rather than this wait loop, it cleaner to just force exchange peer info again?

Reply:
Seems like that's worth trying: https://github.com/holochain/holochain/pull/4834/commits/dbf148d95b5c899e9f1b35fe6745a30eda8748e3

Waiting on CI